contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol

```diff
@@ -0,0 +1,155 @@
+///
+contract ZKSyncValidator is TypeAndVersionInterface, AggregatorValidatorInterface, SimpleWriteAccessController {
+  int256 private constant ANSWER_SEQ_OFFLINE = 1;
+  uint32 private s_gasLimit;
+  uint32 private s_l2GasPerPubdataByteLimit;
+
+  // solhint-disable-next-line chainlink-solidity/prefix-immutable-variables-with-i
+  address public immutable L1_CROSS_DOMAIN_MESSENGER_ADDRESS;
+  // solhint-disable-next-line chainlink-solidity/prefix-immutable-variables-with-i
+  address public immutable L2_UPTIME_FEED_ADDR;
+
+  uint32 private immutable i_chainId;
+  uint32 private constant TEST_NET_CHAIN_ID = 300;
+  uint32 private constant MAIN_NET_CHAIN_ID = 324;
+
+  ///
+  /// @notice emitted when gas cost to spend on L2 is updated
+  /// @param gasLimit updated gas cost
+  ///
+  event GasLimitUpdated(uint32 gasLimit);
+
+  ///
+  /// @param l1CrossDomainMessengerAddress address the Bridgehub contract address
+  /// @param l2UptimeFeedAddr the address of the ZKSyncSequencerUptimeFeedInterface contract address
+  /// @param gasLimit the gasLimit to use for sending a message from L1 to L2
+  ///
+  constructor(
+    address l1CrossDomainMessengerAddress,
+    address l2UptimeFeedAddr,
+    uint32 gasLimit,
+    uint32 chainId,
+    uint32 l2GasPerPubdataByteLimit
+  ) {
+    // solhint-disable-next-line gas-custom-errors
+    require(l1CrossDomainMessengerAddress != address(0), "Invalid xDomain Messenger address");
+    // solhint-disable-next-line gas-custom-errors
+    require(l2UptimeFeedAddr != address(0), "Invalid ZKSyncSequencerUptimeFeedInterface contract address");
+    L1_CROSS_DOMAIN_MESSENGER_ADDRESS = l1CrossDomainMessengerAddress;
+    L2_UPTIME_FEED_ADDR = l2UptimeFeedAddr;
+    s_gasLimit = gasLimit;
+
+    // Check if the chainId is one of the valid values
+    require(chainId == TEST_NET_CHAIN_ID || chainId == MAIN_NET_CHAIN_ID, "Invalid chain id");
+    i_chainId = chainId;
+
+    s_l2GasPerPubdataByteLimit = l2GasPerPubdataByteLimit;
+  }
+
+  ///
+  /// @inheritdoc TypeAndVersionInterface
+  ///
+  function typeAndVersion() external pure virtual override returns (string memory) {
+    return "ZKSyncValidator 1.0.0";
+  }
+
+  ///
+  /// @notice sets the new gas cost to spend when sending cross chain message
+  /// @param gasLimit the updated gas cost
+  ///
+  function setGasLimit(uint32 gasLimit) external onlyOwner {
+    s_gasLimit = gasLimit;
+    emit GasLimitUpdated(gasLimit);
+  }
+
+  ///
+  /// @notice fetches the gas cost of sending a cross chain message
+  ///
+  function getGasLimit() external view returns (uint32) {
+    return s_gasLimit;
+  }
+
+  ///
+  /// @notice sets the l2GasPerPubdataByteLimit TODO complete this
+  /// @param l2GasPerPubdataByteLimit the updated l2GasPerPubdataByteLimit
+  ///
+  function setL2GasPerPubdataByteLimit(uint32 l2GasPerPubdataByteLimit) external onlyOwner {
+    s_l2GasPerPubdataByteLimit = l2GasPerPubdataByteLimit;
+    // emit GasLimitUpdated(gasLimit); TODO do we need to?
```

Comment:
I think it's useful to emit an event here

Let's define a new one called `event GasPerPubdataByteLimitUpdated(uint32 gasLimit);`

That way we can distinguish between the different types of variables that are getting updated